mm2src/coins/rpc_command/get_current_mtp.rs

```diff
@@ -0,0 +1,71 @@
+use common::{HttpStatusCode, StatusCode};
+use derive_more::Display;
+use mm2_core::mm_ctx::MmArc;
+use mm2_err_handle::prelude::MmError;
+
+use crate::{lp_coinfind_or_err,
+            utxo::{rpc_clients::UtxoRpcError, UtxoCommonOps},
+            CoinFindError, MmCoinEnum};
+
+pub type GetCurrentMtpRpcResult<T> = Result<T, MmError<GetCurrentMtpError>>;
+
+#[derive(Deserialize)]
+pub struct GetCurrentMtpRequest {
+    coin: String,
+}
+
+#[derive(Serialize)]
+pub struct GetCurrentMtpResponse {
+    mtp: u32,
+}
+
+#[derive(Serialize, Display, SerializeErrorType)]
+#[serde(tag = "error_type", content = "error_data")]
+pub enum GetCurrentMtpError {
+    NotSupported(String),
+    Internal(String),
+}
+
+impl HttpStatusCode for GetCurrentMtpError {
+    fn status_code(&self) -> StatusCode {
+        match self {
+            GetCurrentMtpError::Internal(_) => StatusCode::INTERNAL_SERVER_ERROR,
+            GetCurrentMtpError::NotSupported(_) => StatusCode::INTERNAL_SERVER_ERROR,
+        }
+    }
+}
+
+impl From<UtxoRpcError> for GetCurrentMtpError {
+    fn from(_: UtxoRpcError) -> Self { GetCurrentMtpError::Internal("Unable to get current mtp".to_string()) }
+}
+
+impl From<CoinFindError> for GetCurrentMtpError {
+    fn from(_: CoinFindError) -> Self { GetCurrentMtpError::Internal("Coin not founded or not activated".to_string()) }
```

Comment:
You should use the `CoinFindError`. Although `CoinFindError` is an internal error, it seems that `GetCurrentMtpError::Internal` will be used only here so it's better to rename it to `GetCurrentMtpError::NoSuchCoin` like similar implementations.